react/Context/context.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+interface ContextStoreType {
+  button: boolean
+  setButton: (button: boolean) => void
+  valueGiftCard: number
+  code: string
+}
+const ContextStore = React.createContext<ContextStoreType>({
+  button: false,
+  setButton: () => {},
+  valueGiftCard: 0,
+  code: '',
+})
+
+export default ContextStore
```

Comment:
Maybe it's a good idea to change the file structure and name! It's common to have the context and the provider in the same file, while also export the hook and provider from the file, like this:
https://github.com/vtex/opt-in-campaign-management/blob/main/react/hooks/useSearchRanges.tsx#L81

Than, instead of context, you would have a hooks folder, that you can import directly into your components without useContext:
https://github.com/vtex/opt-in-campaign-management/blob/93e2eec3c19298f6f0ad57ee6a7f3f9e258079f6/react/components/RangeConfigGrid/index.tsx#L55

If you'd rather have context and provider separated, I'd just change the file name to StoreContext.tsx and the provider to StoreProvider.tsx, and export the hook from the context file!